apps/backoffice/src/lib/be/apim-service.ts

```diff
@@ -42,6 +49,73 @@ export const getApimService: () => ApimUtils.ApimService = cache(() => {
   );
 });
 
+const getAxiosInstance = cache(async () => {
+  const apimConfig = getApimConfig();
+
+  const credential = new ClientSecretCredential(
+    apimConfig.AZURE_CLIENT_SECRET_CREDENTIAL_TENANT_ID,
+    apimConfig.AZURE_CLIENT_SECRET_CREDENTIAL_CLIENT_ID,
+    apimConfig.AZURE_CLIENT_SECRET_CREDENTIAL_SECRET,
+    {
+      authorityHost: AzureAuthorityHosts.AzurePublicCloud
+    }
+  );
+  const tokenResponse = await credential.getToken(
+    apimConfig.AZURE_CREDENTIALS_SCOPE_URL
+  );
+
+  const accessToken = tokenResponse?.token || null;
+
+  if (!accessToken) {
+    throw new Error("invalid or null accessToken");
+  }
+
+  return axios.create({
+    baseURL: apimConfig.AZURE_APIM_SUBSCRIPTIONS_API_BASE_URL,
+    timeout: 5000,
+    headers: { Authorization: `Bearer ${accessToken}` }
+  });
+});
```

Comment:
**question:** this rest client instance is cached, but what if the [AccesToken](https://learn.microsoft.com/en-us/javascript/api/@azure/identity/accesstoken) retrieved by [getToken](https://learn.microsoft.com/en-us/javascript/api/@azure/identity/clientsecretcredential?view=azure-node-latest#@azure-identity-clientsecretcredential-gettoken) during Axios instance creation expired? (currently, I think the SDK handles this automatically)

Reply:
cache is now removed, will follow a PR with a strategy to solve this issue properly